### PR TITLE
is the mypy_pkg_file needed?

### LINKED ISSUE #68: poetry mypy_src step needs to create the pth file
`poetry mypy_src` fails in a clean environment because that step doesn't include the `create_mypy_pkg_file` step

@BrenoJesusFernandes 

Workaround now is to do `poetry test_src`

### PR DESCRIPTION
**Pull request by twoertwein**
closes #68

The dist test run locally fine for me without having to create this file.

### CODE REVIEW

**Comment by Dr-Irv** [comment]
> The dist test run locally fine for me without having to create this file.

The file is created for the source test.  Put a typing bug in a test, and you'll see it is not found because the local copy of `pandas-stubs` is not used.

Here are the possible ways for the local copy to be tested:
1) change name of directory to `typings/pandas` and modify `pyproject.toml` to tell `mypy` to look there.
2) Install the local copy (but that is what `dist` does)
3) Create a `.pth` file so it knows to look in the current directory.

**Comment by BrenoJesusFernandes** [comment]
tests are passing maybe because the cache from previous builds.

**Comment by twoertwein** [comment]
> The file is created for the source test. Put a typing bug in a test, and you'll see it is not found because the local copy of `pandas-stubs` is not used.

It sounds as this is only needed for local development, or? I installed the distribution, and then added a bug in the local copy, ran mypy, and it found the issue. Do you mind posting the steps which create the issue that you observe?

**Comment by twoertwein** [comment]
> tests are passing maybe because the cache from previous builds.

no longer caching the venv, just the poetry.lock file

**Comment by twoertwein** [comment]
> ```shell
> mypy tests/test_frame.py
> ```

I can confirm that I get errors then, but I usually never run `mypy <specific file>`, I always run `mypy tests/ pandas-stubs/` which works fine even in that case.

**Comment by twoertwein** [comment]
How about we add `files = ["pandas-stubs", "tests"]` to pyproject.toml, and run just `mypy` (no file arguments are needed)?

I don't think it is worth having this workaround on the CI just to make `mypy tests/test_frame.py` work, mypy is very fast on pandas-stubs and the tests.

**Comment by BrenoJesusFernandes** [comment]
I am feeling we are missing something ... when I arrive at my machine I am gonna test again to see if the behaviour is the same.

**Comment by twoertwein** [comment]
Do you have a refrence to an existing mypy issue about this? I think it could be a mypy bug, if we add `mypy_path="."` it should always find pandas-stubs, but it doesn't!

**Comment by Dr-Irv** [comment]
See this:
https://mypy.readthedocs.io/en/stable/installed_packages.html#installed-packages

> Starting in mypy 0.900, most third-party package stubs must be installed explicitly.

And this:
https://mypy.readthedocs.io/en/stable/stubs.html

> Note that [PEP 561](https://peps.python.org/pep-0561/) stub-only packages must be installed, and may not be pointed at through the MYPYPATH

Also, `mypy tests` doesn't work either.

**Comment by twoertwein** [comment]
Thank you @Dr-Irv that explains why it isn't working. Then how about:

- only testing the `pandas-stubs` (locally) with mypy and pyright
- install `pandas-stubs`
- and then test `tests` with mypy and pyright using the installed stubs?

**Comment by Dr-Irv** [comment]
No. Because in development, it is really useful (and it saves time) to run the tests without building a dist and installing them.

I have had times where the tests pass `pyright`, but not `mypy`, so it is useful to just run `mypy` by itself.

**Comment by twoertwein** [comment]
You can always run `mypy pandas-stubs/ tests/` without having to install anything (for some reason that works).

File: scripts/test/__init__.py [hunk 1]
```diff
@@ -76,9 +61,6 @@ def test_all(clean_cache: bool = False):
 
     steps.extend(
         [
-            _step.create_mypy_pkg_file,
-            _step.mypy_src,
```
Comment (Dr-Irv) [inline scripts/test/__init__.py:80]: can you keep this step in there?

File: scripts/test/__init__.py [hunk 2]
```diff
@@ -13,26 +13,11 @@ def test_src(profile: str, clean_cache: bool = False):
...
-            ]
-        )
+        steps.extend([_step.mypy_src, _step.pyright_src])
     elif profile == "pytest":
         steps.extend([_step.pytest_src])
     elif profile == "full":
-        steps.extend(
-            [
-                _step.create_mypy_pkg_file,
-                _step.mypy_src,
```
Comment (Dr-Irv) [inline scripts/test/__init__.py:30]: can you keep the mypy step in there?

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
So since `mypy tests pandas-stubs` works on the local source, let's keep that in there so you can just test locally.

File: scripts/test/__init__.py (hunk 1 above)
Comment (twoertwein) [reply scripts/test/__init__.py:80]: Thank you, I removed that line accidentally.

**Review by Dr-Irv — APPROVED** [review]
thanks @twoertwein

**Comment by BrenoJesusFernandes** [comment]
Indeed is working without the file in my setup, but I'm still curious why? haha

**Pull request merged**